### PR TITLE
Jtorre39/ai health damage

### PR DESCRIPTION
**Pull request by im-deepfriedwater**
### Includes:
- Implementation for zombies (tracking, attacking player & walls)
- Implementation for a health system ui
- Implementation for getting hit (knockback, invincibility)
- Tweaks to the Invector scripts
- A quick bake of the navmesh on the map prototype so zombies can walk on it
- death animation and game over screen upon zero health

### CODE REVIEW

**Comment by im-deepfriedwater** [comment]
There's a few changes that were from the sandbox prototypes, I'll mark them accordingly and you can ignore those.

File: SandBox/EncounterPrototype/EncounterPrototype/Assets/Invector-3rdPersonController/Basic Locomotion LITE/Scripts/CharacterController/vThirdPersonInput.cs [hunk 1]
```diff
@@ -63,7 +63,7 @@ protected virtual void LateUpdate()
 
         protected virtual void FixedUpdate()
         {
```
Comment (im-deepfriedwater) [inline SandBox/EncounterPrototype/EncounterPrototype/Assets/Invector-3rdPersonController/Basic Locomotion LITE/Scripts/CharacterController/vThirdPersonInput.cs:65]: ignore, from sandbox prototype

File: SandBox/EncounterPrototype/EncounterPrototype/Assets/Invector-3rdPersonController/Basic Locomotion LITE/Scripts/CharacterController/vThirdPersonMotor.cs [hunk 2]
```diff
@@ -10,7 +10,7 @@ namespace Invector.CharacterController
     public abstract class vThirdPersonMotor : MonoBehaviour
     {
         #region Variables        
-
```
Comment (im-deepfriedwater) [inline SandBox/EncounterPrototype/EncounterPrototype/Assets/Invector-3rdPersonController/Basic Locomotion LITE/Scripts/CharacterController/vThirdPersonMotor.cs:13]: ignore, from sandbox prototype

File: SandBox/EncounterPrototype/EncounterPrototype/Assets/Scripts/PlayerBehaviour.cs [hunk 3]
```diff
@@ -21,10 +21,13 @@ void Start()
 
     void Update()
     {
```
Comment (im-deepfriedwater) [inline SandBox/EncounterPrototype/EncounterPrototype/Assets/Scripts/PlayerBehaviour.cs:23]: ignore, from sandbox prototype

File: SandBox/EncounterPrototype/EncounterPrototype/Assets/Scripts/SphereTest.cs [hunk 4]
```diff
@@ -0,0 +1,19 @@
+﻿using System.Collections;
+using System.Collections.Generic;
```
Comment (im-deepfriedwater) [inline SandBox/EncounterPrototype/EncounterPrototype/Assets/Scripts/SphereTest.cs:2]: ignore, from sandbox prototype

File: SandBox/EncounterPrototype/EncounterPrototype/ProjectSettings/ProjectVersion.txt [hunk 5]
```diff
@@ -1 +1 @@
-m_EditorVersion: 2018.3.3f1
```
Comment (im-deepfriedwater) [inline SandBox/EncounterPrototype/EncounterPrototype/ProjectSettings/ProjectVersion.txt:1]: ignore, from sandbox prototype

File: unity-ktwo/Assets/Scripts/Behaviours/WallBehaviour.cs [hunk 6]
```diff
@@ -0,0 +1,12 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class WallBehaviour: MonoBehaviour
+{
+    public void Break () // Should be called when there is 0 health left;
+    {
+        Debug.Log("Lol");
```
Comment (CapnSquirrel) [inline unity-ktwo/Assets/Scripts/Behaviours/WallBehaviour.cs]: remove this

File: unity-ktwo/Assets/Scripts/Controllers/EnemyController.cs [hunk 7]
```diff
@@ -0,0 +1,172 @@
...
+            isAttackOnCooldown = true;
+            if (StartAttackCooldownCourtineStarted)
+            {
+                return;
+            }
+            StartCoroutine("StartAttackCooldown");
+        } else if (isAttacking && !isAttackOnCooldown && hitboxActivated && other.gameObject.tag == "Player")
+        {
+            hasAttacked = true;
+            Debug.Log("tryn");
```
Comment (CapnSquirrel) [inline unity-ktwo/Assets/Scripts/Controllers/EnemyController.cs]: remove this

File: unity-ktwo/Assets/Scripts/Controllers/EnemyController.cs [hunk 8]
```diff
@@ -0,0 +1,172 @@
...
+    }
+
+    IEnumerator CalculateAttackCooldown ()
+    {   
+        float currentTime = 0;
+        while (currentTime <= attackCooldown)
+        {
+            currentTime += Time.deltaTime;
+            yield return null;
+        }
```
Comment (CapnSquirrel) [inline unity-ktwo/Assets/Scripts/Controllers/EnemyController.cs]: yield return new WaitForSeconds(attackCooldown)

**Review by CapnSquirrel — APPROVED** [review]

**Pull request merged**